logs.go

```diff
@@ -91,6 +92,23 @@ func getLoggableResources(getPodContainers func(string, string) ([]string, error
 	return loggableResources, nil
 }
 
+// GetProjectPodsBySubstr returns a list of pod names that contain the provided string, in the named project.
+func getProjectPodNamesBySubstr(project, substr string) ([]string, error) {
+	ret := []string{}
+	pods, err := GetResourceNames(project, "pod")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pod := range pods {
+		if strings.Contains(pod, substr) {
+			ret = append(ret, pod)
+		}
+	}
+
+	return ret, nil
```

Comment:
@vchepeli The philosophy here is to exit early, rather than saving up errors until the end of the function. If an error is discovered then it should be reported and the function should exit because if an error is found it would be unwise to keep going in a known to be unstable situation.

To keep these advantages and only have one return statement would mean nesting if statements around every possible error scenario, which is much less readable.